consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -711,40 +718,280 @@ impl PruningProofManager {
         Err(PruningImportError::PruningProofNotEnoughHeaders)
     }
 
+    // TODO: Find a better name
+    fn find_current_dag_level(&self, pp_header: &Header) -> BlockLevel {
+        let direct_parents = BlockHashSet::from_iter(pp_header.direct_parents().iter().copied());
+        pp_header
+            .parents_by_level
+            .iter()
+            .enumerate()
+            .skip(1)
+            .find_map(|(level, parents)| {
+                if BlockHashSet::from_iter(parents.iter().copied()) == direct_parents {
+                    None
+                } else {
+                    Some((level - 1) as BlockLevel)
+                }
+            })
+            .unwrap_or(self.max_block_level)
+    }
+
+    fn estimated_blue_depth_at_level_0(&self, level: BlockLevel, level_depth: u64, current_dag_level: BlockLevel) -> u64 {
```

Comment:
What is the reasoning behind this estimate?

Why should levels lower than current DAG level require more depth?

I was thinking it would make sense for it to be `level_depth << level.saturating_sub(current_dag_level).min(10)`  because there would be less blocks in levels higher than the current dag level so when searching you would like to start somewhere deeper.